assets/blocks/course-outline/lesson-block/lesson-edit-toolbar.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	Button,
+	ExternalLink,
+	Spinner,
+	Toolbar,
+	ToolbarItem,
+} from '@wordpress/components';
+import { dispatch, useSelect } from '@wordpress/data';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { COURSE_STORE } from '../course-outline-store';
+
+const getLessonURL = ( lessonId ) => `post.php?post=${ lessonId }&action=edit`;
+
+/**
+ * Link to edit the Lesson in a new tab.
+ *
+ * @param {Object} props          Component props.
+ * @param {number} props.lessonId The lesson ID.
+ */
+export const EditLessonLink = ( { lessonId } ) => (
+	<ExternalLink
+		href={ getLessonURL( lessonId ) }
+		target="lesson"
+		className="wp-block-sensei-lms-course-outline-lesson__edit"
+	>
+		{ __( 'Edit lesson', 'sensei-lms' ) }
+	</ExternalLink>
+);
+
+/**
+ * Toolbar section for the link to edit a lesson.
+ *
+ * @param {Object} props          Component props.
+ * @param {number} props.lessonId The lesson ID.
+ */
+const LessonEditToolbar = ( { lessonId } ) => {
+	// Determine whether we are currently saving.
+	const { isSavingPost, isSavingMetaBoxes, isSavingStructure } = useSelect(
+		( select ) => ( {
+			isSavingPost: select( 'core/editor' ).isSavingPost(),
+			isSavingMetaBoxes: select( 'core/edit-post' ).isSavingMetaBoxes(),
+			isSavingStructure: select( COURSE_STORE ).getIsSavingStructure(),
+		} )
+	);
+
+	// Component for the "Save and edit lesson" button.
+	const savePostLink = (
+		<ToolbarItem
+			as={ Button }
+			onClick={ () => {
+				dispatch( 'core/editor' ).savePost();
```

Comment:
I think the most recommended way in the components would be using the `useDispatch` hook. I don't have a strong reason for that, but if we check Gutenberg code, for example, `dispatch` would mostly be used in other contexts out of React.

An old discussion about that: p1603303892144000-slack-C45SNKV4Z

Reply:
Done in 3c03ad7ef2c20f25bc9599b15b62abf99ab94452